### DISCUSSION TITLE
Extract pore network using SNOW with pore.inscribed_diameter > throat.inscribed_diameter

### DISCUSSION THREAD

**Discussion by MTJKoning** [PMEAL/porespy#908] [Q&A]
Hi professor,

I am implementing a non-linear transport model in OpenPNM (where the hydraulic conductance is dependent on flow rate), which works, but I need to make sure that my pore radius is larger than my throat radius in order to avoid nan values. My question is:

Is it possible to extract a pore network using the SNOW algorithm, where for every conduit the pore.inscribed_diameter > throat.inscribed_diameter? If yes, how? Or do I need to use the MaxBall algorithm for that? 

Kind regards,

Menno Koning

**Comment by jgostick** [comment]
It is not possible to guarantee this in the extraction, but you can adjust the sizes of your throats during a post-processing step once it's imported into OpenPNM.  Basically just add the "from_neighbor_pores" pore scale model, with 'pore.diameter' as the `prop` and 'min' as the `mode`.  You'll want to store the result in a back-up array so you don't over write the original throat size.  Then you can check to see where the original throat size is larger than the one computed by the model, and overwrite as needed. This would also let you see (a) how many throats you needed to change, and (b) by how much.  Hopefully it is not too many.  

The function would be something like:
```python
pn.add_model(
    propname='throat.diameter_max',
    model=op.models.geometry.throat_size.from_neighbor_pores,
    mode='min',
)
```

**Reply by jgostick** [reply]
I should also add that you should use the 'pore.extended_diameter' instead of 'pore.inscribed_diameter', which will help quite a bit.
